mParticle-Apple-SDK/Network/MPNetworkCommunication.m

```diff
@@ -114,6 +114,31 @@ - (instancetype)init {
 }
 
 #pragma mark Private accessors
+
+- (NSString *)defaultHostWithSubdomain:(NSString *)subdomain apiKey:(NSString *)apiKey enableDirectRouting:(BOOL)enableDirectRouting {
+    if (enableDirectRouting) {
+        NSArray *splitKey = [apiKey componentsSeparatedByString:@"-"];
```

Comment:
What would happen if the subdomian doesnt have the - for splitting?

Reply:
if the API key doesn't have a dash (aka it's an old key without a prefix) this will just return an array with a single value, which I check for on the next line.